portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,33 +36,50 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-
-    public ArrayList<String> messages;
+  public ArrayList<String> messages = new ArrayList<String>();
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // below command was used in servlet and fetch() tutorials
-    // response.setContentType("text/html;");
-    //
-    // below command was used in the servlet tutorial
-    // response.getWriter().println("<h1>Hello Gracie!</h1>");
-    //
-    // below command was used in the fetch() tutorial
-    // response.getWriter().println("Hello, Gracie is testing the fetch() function!");
-    //
     messages = new ArrayList<String>();
-    messages.add("Hello");
-    messages.add("Hi");
-    messages.add("Greetings");
-
-    String json = convertToJsonUsingGson(messages);
+    Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    for (Entity entity : results.asIterable()) {
+      messages.add(entity.getProperty("content").toString());
+    }
+    String json_comments = convertToJsonUsingGson(messages);
     response.setContentType("application/json;");
-    response.getWriter().println(json);
+    response.getWriter().println(json_comments);
+  }
+
+  public String convertToJsonUsingGson(ArrayList<String> message_arraylist) {
+    Gson gson = new Gson();
+    String json = gson.toJson(message_arraylist);
+    return json;
   }
 
-  public String convertToJsonUsingGson(ArrayList<String> m) {
-      Gson gson = new Gson();
-      String json = gson.toJson(m);
-      return json;
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Retrieve inputs
+    String text = request.getParameter("text-input");
+    long time = System.currentTimeMillis();
+    
+    // Prints result retrieved from sentiment analysis
+    Document doc = Document.newBuilder().setContent(text).setType(Document.Type.PLAIN_TEXT).build();
+    LanguageServiceClient languageService = LanguageServiceClient.create();
+    Sentiment sentiment = languageService.analyzeSentiment(doc).getDocumentSentiment();
+    float score = sentiment.getScore();
+    languageService.close();
+    System.out.println("Score:" + score);
```

Comment:
nit (suggestion but completely optional): instead of printing out, what about displaying the score next to each comment by setting is as a textEntity property?

Reply:
This was created in the next pull request.